### PR TITLE
Bump spead2 version to 4.0

### PR DESCRIPTION
**Pull request by james-smith-za**
Bump spead2 to version 4.0

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use sphinx-apidoc to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-1075.

### CODE REVIEW

File: qualification/requirements.txt [hunk 1]
```diff
@@ -203,7 +203,6 @@ numpy==1.23.5
     #   pandas
     #   prometheus-api-client
     #   scipy
-    #   spead2
```
Comment (bmerry) [inline qualification/requirements.txt:206]: Oh dear, it looks like spead2 has a typo in pyproject.toml that makes it not actually depend on numpy. I probably need to release a bugfix release. I think it shouldn't have any bad effects on katgpucbf for now though.

File: Dockerfile [hunk 2]
```diff
@@ -87,11 +87,10 @@ RUN SPEAD2_VERSION=$(grep ^spead2== katgpucbf/requirements.txt | cut -d= -f3) &&
     wget "https://github.com/ska-sa/spead2/releases/download/v$SPEAD2_VERSION/spead2-$SPEAD2_VERSION.tar.gz" && \
     tar -zxf "spead2-$SPEAD2_VERSION.tar.gz" && \
     cd "spead2-$SPEAD2_VERSION" && \
-    mkdir build && \
+    meson setup build && \
```
Comment (bmerry) [inline Dockerfile:90]: I suggest changing this to
```suggestion
    meson setup -Dtools=disabled -Dibv=enabled -Dmlx5dv=enabled build && \
```
That will
- force the build to fail if ibv or mlx5dv is not found (in which case the rest of the Dockerfile hasn't installed things properly)
- skip building the command-line tools, which aren't needed for fsim

Then you should be able to remove boost-program-options-dev.

**Review by bmerry — CHANGES_REQUESTED** [review]
Any idea what's up with the failed Jenkins build? I've not seen that sort of error before.

**Comment by james-smith-za** [comment]
> Any idea what's up with the failed Jenkins build? I've not seen that sort of error before.

I have no idea. It's a mystery to me.
I made a new branch with a trivial change just to see if it gets stuck at the same point and it didn't.

**Comment by james-smith-za** [comment]
Bruce if you can cut a release for 4.0.1 then I'll include it in this PR.

**Comment by bmerry** [comment]
> Bruce if you can cut a release for 4.0.1 then I'll include it in this PR.

Will do, but I might only get it done on Monday.

**Comment by bmerry** [comment]
> Bruce if you can cut a release for 4.0.1 then I'll include it in this PR.

Done.

File: Dockerfile [hunk 3]
```diff
@@ -59,9 +59,9 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-ins
     python-is-python3 \
     git \
     pkg-config \
+    ninja-build \
     libboost-dev \
     libboost-program-options-dev \
```
Comment (bmerry) [inline Dockerfile:64]: Are you able to remove this now that you're passing `-Dtools=disabled`?

File: Dockerfile [hunk 4]
```diff
@@ -76,7 +76,7 @@ RUN python -m venv /venv
 ENV PATH=/venv/bin:$PATH
 # Install up-to-date versions of installation tools, for the benefits of
 # packages not using PEP 517/518.
```
Comment (bmerry) [inline Dockerfile:78]: This comment needs to be updated to explain why jinja etc are installed: it's for compiling the C++ bindings of spead2.

**Review by bmerry — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (james-smith-za) [reply Dockerfile:64]: If I remove that line, I stiill get the "ERROR: Could not detect Ninja v1.8.2 or newer" error.

Comment (bmerry) [reply Dockerfile:64]: I think you're looking at the wrong line. I'm talking about libboost-program-options-dev.

File: Dockerfile [hunk 5]
```diff
@@ -76,7 +76,9 @@ RUN python -m venv /venv
...
-RUN pip install pip==22.3.1 setuptools==65.6.3 wheel==0.38.4
+# jina2, pypackagin, pycparser and meson are installed in order to compile the
```
Comment (bmerry) [inline Dockerfile:79]:
```suggestion
# jinja2, packaging, pycparser and meson are installed in order to compile the
```

**Review by bmerry — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (james-smith-za) [reply Dockerfile:64]: Ah. Let me give that a try quickly.

Comment (james-smith-za) [reply Dockerfile:64]: The fsim still seems to need it.

**Review by bmerry — APPROVED** [review]

**Pull request merged**